File: services/shop-backend/src/main/java/io/miragon/shop/adapter/outbound/persistence/order/OrderJpaRepository.java
package io.miragon.shop.adapter.outbound.persistence.order;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface OrderJpaRepository extends JpaRepository<OrderEntity, UUID> {
    List<OrderEntity> findByUserId(String userId);
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/adapter/outbound/persistence/order/OrderJpaRepository.kt
package io.miragon.shop.adapter.outbound.persistence.order

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface OrderJpaRepository : JpaRepository<OrderEntity, UUID> {
    fun findByUserId(userId: String): List<OrderEntity>
}